raiden-cli/src/routes/pendingTransfers.ts

```diff
@@ -1,20 +1,61 @@
 import { Router, Request, Response } from 'express';
+import { timer } from 'rxjs';
+import { filter, takeUntil, map, toArray } from 'rxjs/operators';
+import { RaidenTransfer, isntNil } from 'raiden-ts';
+
 import { Cli } from '../types';
+import { validateOptionalAddressParameter } from '../utils/validation';
 
-export function makePendingTransfersRouter(this: Cli): Router {
-  const router = Router();
+enum ApiRole {
+  initiator = 'initiator',
+  mediator = 'mediator',
+  target = 'target,',
+}
 
-  router.get('/', function (_request: Request, response: Response): void {
-    response.status(404).send('Not implemented yet');
-  });
+function transformSdkToApiPendingTransfer(this: Cli, transfer: RaidenTransfer) {
+  let role: ApiRole;
+  if (transfer.direction === 'sent' && transfer.initiator === this.raiden.address)
+    role = ApiRole.initiator;
+  else if (transfer.direction === 'received' && transfer.target === this.raiden.address)
+    role = ApiRole.target;
+  else if (transfer.direction === 'sent') role = ApiRole.mediator;
+  else return; // filter out duplicated 'received' side of mediated transfers
+  return {
```

Comment:
Usually we had an interface defined for these objects that get shared via the API. Either remove the old ones or extend here please.

Reply:
Checkout conversations above.